### PR TITLE
[MDS-5239] Converting rest of Mine Permit to Typescript

### PR DESCRIPTION
**Pull request by asinn134**
## Objective 

[MDS-5239](https://bcmines.atlassian.net/browse/MDS-5239)

- Converts MinePermitCard, PermitAmendmentFileUpload, PermitAmendmentUploadedFilesList, PermitConditionManagement, NOWActionWrapper, and AddCondition components to Typescript

- Converted PermitConditionManagement, NOWActionWrapper and AddCondition to functional components

- Sets the typing for the rest of Permit ActionCreators

- Converts the tests for the above components

### CODE REVIEW

File: services/core-web/src/components/mine/Permit/PermitAmendmentUploadedFilesList.tsx [hunk 1]
```diff
@@ -28,6 +28,7 @@ export const PermitAmendmentUploadedFilesList = (props) => (
                 <p>Are you sure you want to remove this file?</p>,
                 <p>This cannot be undone.</p>,
               ]}
+              /* eslint react/jsx-key: 0 */
```
Comment (taraepp) [inline services/core-web/src/components/mine/Permit/PermitAmendmentUploadedFilesList.tsx]: Are these of type IPermitAmendmentMineDocument? Can we use that to assign a key?
If not, eslint-disable-next-line is a little easier to search

Comment (matbusby-fw) [reply services/core-web/src/components/mine/Permit/PermitAmendmentUploadedFilesList.tsx]: :100:

File: services/common/src/interfaces/dropDownListItem.interface.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface IDropDownListItem {
```
Comment (matbusby-fw) [inline services/common/src/interfaces/dropDownListItem.interface.ts]: There is already an IOption interface here `services/common/src/interfaces/common/option.interface.ts`.  The only difference is that the value is `string`, but we can change it to `string | number` without any problem.

**Review by matbusby-fw — CHANGES_REQUESTED** [review]
Looks awesome!  Just the one comment about a duplicate dropDownItem interface, and Tara's suggestion I think.  For the drop down, it doesn't matter which one we keep as long as we update wherever they're being used.

File: services/core-web/src/components/mine/Permit/PermitAmendmentUploadedFilesList.tsx (hunk 1 above)
Comment (asinn134) [reply services/core-web/src/components/mine/Permit/PermitAmendmentUploadedFilesList.tsx]: This has been adjusted to include keys now. I honestly thought we only needed to add a key in the parent element in React mappings, so I was really confused about the error I was getting, but yeah adding keys to the paragraph elements solves this issue, thanks!

File: services/common/src/interfaces/dropDownListItem.interface.ts (hunk 2 above)
Comment (asinn134) [reply services/common/src/interfaces/dropDownListItem.interface.ts]: `IOption` has been adjusted now and `IDropDownListItem ` removed

**Review by matbusby-fw — APPROVED** [review]
👍

**Review by taraepp — APPROVED** [review]
👍

**Pull request merged**